examples/TUTORIAL.md

```diff
@@ -0,0 +1,253 @@
+    types_hash_max_size 2048;
+
+    include             /etc/nginx/mime.types;
+    default_type        application/octet-stream;
+
+    # Load modular configuration files from the /etc/nginx/conf.d directory.
+    # See http://nginx.org/en/docs/ngx_core_module.html#include
+    # for more information.
+    include /opt/app-root/etc/conf.d/*.conf;
+
+    server {
+        listen       8080 default_server;
+        server_name  _;
+        root         /opt/app-root/src; # tell nginx to load the main html application files from our /opt/app-root/src folder.
+
+        # Load configuration files for the default server block.
+        include /opt/app-root/etc/default.d/*.conf;
+
+        location / {
+        }
+
+        error_page 404 /404.html;
+            location = /40x.html {
+        }
+
+        error_page 500 502 503 504 /50x.html;
+            location = /50x.html {
+        }
+    }
+}
+```
+In our assemble script we define other actions that need to be ran when we want to install our application, this is where we build python modules or bundle install our required components, however in this example we leave it as it is. It will only copy the source files from `/tmp/src` to our `/opt/app-root/src`.
+```bash
+#!/bin/bash -e
+#
+# S2I assemble script for the 'nginx-centos7' image.
+# The 'assemble' script builds your application source ready to run.
+#
+# For more information refer to the documentation:
+#	https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md
+#
+
+if [ "$1" = "-h" ]; then
+	# If the 'nginx-centos7' assemble script is executed with '-h' flag,
+	# print the usage.
+	exec /usr/libexec/s2i/usage
+fi
+
+# Restore artifacts from the previous build (if they exist).
+#
+if [ "$(ls /tmp/artifacts/ 2>/dev/null)" ]; then
+  echo "---> Restoring build artifacts"
+  mv /tmp/artifacts/. ./
+fi
+
+echo "---> Installing application source"
+cp -Rf /tmp/src/. ./
+
+```
+
+There's also a usage script that lets the user know how he can run the intended image.
+
+What we need to do is tell the run script what process to run on start.
+
+```bash
+#!/bin/bash -e
+#
+# S2I run script for the 'nginx-centos7' image.
+# The run script executes the server that runs your application.
+#
+# For more information see the documentation:
+#	https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md
+#
+
+# We will turn off daemonizing for the nginx process so that the container
+# doesn't exit after the process runs.
+# We will also specify our configuration that we copied in the builder image.
+exec /usr/sbin/nginx -c /opt/app-root/etc/nginx.conf -g "daemon off;"
+```
+Next, we continue by creating our base builder image with make.
+
+In the current project directory (`source-to-image/examples/s2i-nginx`) just issue the `make` command. Make creates our base builder image from the instructions in our `Dockerfile`. After it finishes we can use our builder image to test out the webserver configuration.
```

Comment:
Assume a user is following you step by step, which means you should not link to examples directory, but rather provide them with commands he can easily follow on their own. Again have a look at my article. I'd point a user to it at the beginning or the end of the article, there's a accompanying repo one might be interested in.